### DISCUSSION TITLE
how to defer js files generated by the docusaurus 2.xx

### DISCUSSION THREAD

**Discussion by numankhan4** [facebook/docusaurus#9799] [Q&A]
I've extensively customized Docusaurus theme components in my project using Swizzle wrap, and I'm currently using Docusaurus 2.xx. I'm aware that the defer issue has been resolved in Docusaurus 3.xx. However, due to the extensive customization in my project, I'm reluctant to upgrade to Docusaurus 3.

Request for Assistance:
I would like to defer the bundled JavaScript files generated by Docusaurus. Specifically, I'm looking to defer the following files:

runtime~main.xx.js
main.xx.js
If anyone has experience or suggestions on how to achieve this within the context of Docusaurus 2.xx, I would greatly appreciate your guidance.

Thank you in advance for your help!

**Comment by slorber** [comment]
What do you mean by the "defer issue". 

I don't know what you mean with that term, and why you think we fixed this on v3.

**Reply by numankhan4** [reply]
@slorber I am talking about moving the script into header + defer them. Below is the reference of pull request for your reference. 
https://github.com/facebook/docusaurus/pull/8081

**Reply by slorber** [reply]
We don't support that for v2.x and afaik there's no workaround. 
Now you could technically post-process all the html files to add that attribute manually with a script.
